packages/block-editor/src/components/block-variation-transforms/index.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { SelectControl } from '@wordpress/components';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { useState } from '@wordpress/element';
+
+function __experimentalBlockVariationTransforms( {
+	blockName,
+	selectedBlockClientId,
+} ) {
+	const [ variationsSelectValue, setVariationsSelectValue ] = useState( '' );
+	const { updateBlockAttributes } = useDispatch( 'core/block-editor' );
+	const { variations } = useSelect(
+		( select ) => {
+			const { getBlockVariations } = select( 'core/blocks' );
+			const scope = 'transform';
+			const _variations = getBlockVariations(
+				blockName,
+				scope
+			).filter( ( variation ) => variation.scope?.includes( scope ) );
```

Comment:
It definitely should be handled inside the `getBlockVariations` implementation.

Reply:
This is here because of my comment for the `getBlockVariations`. 

>Do we really need a new option there? Wouldn't it be enough to set the list of scopes to [ 'block', 'inserter' ] if the scope param is not provided?

Where? In `getBlockVariations`? If yes it still feels that the abstraction will not be good to me to have specific business logic per `scope` passed. If in the future we add another `scope` that needs to be filtered exclusively we should add logic for this as well there?

I think this:
```
return variations.filter( ( variation ) => {
		return ! variation.scope || variation.scope.includes( scope );
	} );
```
was there for convenience because `block` as a scope wasn't affecting anything else.

Reply:
How about this:
```diff
diff --git a/packages/blocks/src/store/selectors.js b/packages/blocks/src/store/selectors.js
index 21e88311b5..34bd862e89 100644
--- a/packages/blocks/src/store/selectors.js
+++ b/packages/blocks/src/store/selectors.js
@@ -90,7 +90,11 @@ export function getBlockVariations( state, blockName, scope ) {
 		return variations;
 	}
 	return variations.filter( ( variation ) => {
-		return ! variation.scope || variation.scope.includes( scope );
+		return (
+			( ! variation.scope &&
+				[ 'block', 'inserter' ].includes( scope ) ) ||
+			variation.scope.includes( scope )
+		);
 	} );
 }
```

Would it work?

Reply:
Yes a concept like that would work (I didn't test but though something similar to implement at first), but 
>it still feels that the abstraction will not be good to me to have specific business logic per scope passed. If in the future we add another scope that needs to be filtered exclusively we should add logic for this as well there?

Do you think adding these kind of checks kind of making the abstraction feel wrong?

Reply:
In what way wrong? It only ensures backward compatibility, otherwise, all newly added scopes will be only matched when explicitly listed in the list passed in `scope` field. Do I miss anything here? An example where it won't work as intended would be very helpful.

Reply:
I never said it wouldn't work. It will. I just said that the abstraction didn't seem right to me now that we have more scopes (this `! variation.scope`). Backward compatibility is a thing though... I will make that change as you suggested.

Reply:
I simply didn't take into account that defaulting to all scopes when no scopes are set would cause troubles. I think it all started with `block` as the only supported scope, then we added `inserter`.